### PR TITLE
test: useDocument hook

### LINKED ISSUE #1366: test(dmt-core): useDocument hook

### PR DESCRIPTION
**Pull request by rikkebl**
## What does this pull request change?
Added a new test-file to test useDocument Hook and a new utils-test file.

Missing test on UpdateDocument function.

## Why is this pull request needed?
To test useDocument hook. 

## Issues related to this change
Closes #1366.

### CODE REVIEW

**Comment by eoaksnes** [comment]
@rikkebl What if we inside the mockGetDocument use find to search for document based on dataSourceId and documentId, and return the document if found. If it is not found, error is returned. Then you can remove the shouldFaild input argument.

File: web/packages/dmt-core/src/utils/test-utils-dmt-core.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// @ts-nocheck
+
+import { DmssAPI } from '../services/api/DmssAPI'
+import React from 'react'
+
+export const mockGetDocument = (result: Array<any>, shouldFail: boolean) => {
```
[the remark this answers is not in the document]
Comment (netr0m) [reply web/packages/dmt-core/src/utils/test-utils-dmt-core.tsx:6]: Could we use `TValidEntity` here?

**Review by netr0m — COMMENTED** [review]
Nice! Consider what @soofstad mentioned. Looks good otherwise!

**Review by eoaksnes — APPROVED** [review]
Looks good!

**Pull request merged**